Base/QTGUI/qSlicerModuleSelectorToolBar.cxx

```diff
@@ -265,9 +265,19 @@ void qSlicerModuleSelectorToolBar::onModuleSelected(const QString& name)
 void qSlicerModuleSelectorToolBar::actionSelected(QAction* action)
 {
   Q_D(qSlicerModuleSelectorToolBar);
+  if (action == 0)
+    {
+    // Because the NoModuleAction is not observed by ctkMenuComboBox, the
+    // toolbar shall clear the text of the current action manually.
+    d->ModulesComboBox->clearActiveAction();
+    }
   QAction* lastAction = d->lastSelectedAction();
   if (action == lastAction)
     {
+    // In case the previous action was the NoModuleAction, it would not
```

Comment:
It's not very nice that we keep emitting moduleSelected signals even when the module is not changed. Have you considered making ctkMenuComboBox smarter instead of adding workarounds in this class?